unleashandroidsdk/src/main/java/io/getunleash/android/polling/UnleashFetcher.kt

```diff
@@ -30,54 +37,55 @@ import okhttp3.OkHttpClient
 import okhttp3.Request
 import okhttp3.Response
 import okhttp3.internal.closeQuietly
-import java.io.Closeable
-import java.io.IOException
-import java.util.concurrent.TimeUnit
-import java.util.concurrent.atomic.AtomicReference
-import kotlin.coroutines.CoroutineContext
-import kotlin.coroutines.resume
-import kotlin.coroutines.resumeWithException
 
 /**
- * Http Client for fetching data from Unleash Proxy.
- * By default creates an OkHttpClient with readTimeout set to 2 seconds and a cache of 10 MBs
- * @param httpClient - the http client to use for fetching toggles from Unleash proxy
+ * Http Client for fetching data from Unleash Proxy. By default creates an OkHttpClient with
+ * readTimeout set to 2 seconds and a cache of 10 MBs
+ * @param httpClient
+ * - the http client to use for fetching toggles from Unleash proxy
  */
 open class UnleashFetcher(
-    unleashConfig: UnleashConfig,
-    private val httpClient: OkHttpClient,
-    private val unleashContext: StateFlow<UnleashContext>,
+        unleashConfig: UnleashConfig,
+        private val httpClient: OkHttpClient,
+        private val unleashContext: StateFlow<UnleashContext>,
 ) : Closeable {
     companion object {
         private const val TAG = "UnleashFetcher"
     }
-
+    private var contextForLastFetch: UnleashContext? = null
```

Comment:
Hold the last fetched so we can ignore refreshing toggles when context has change due to a sync setContext which avoids 2 fetch calls for the same context